src/interop.ts

```diff
@@ -58,12 +58,31 @@ export function normalize(value: any, cfnType?: string, pulumiProvider?: PulumiP
     return result;
 }
 
-export type ResourceMapping =
-    | {
-          resource: pulumi.Resource;
-          attributes: { [name: string]: pulumi.Input<any> };
-      }
-    | pulumi.Resource;
+/**
+ * Use this type if you need to control the attributes that are available on the
```

Comment:
Nice to have the doc comment!

This is getting a little bit more intricate. Complexity in the exported API is a cost but looks like we need this, perhaps reasonable.

Looks like remapCloudControlResource needs updating in README.md as this is changing a bit right?

In general I would love if I could easily tell which change affects the user-facing API, vs what is an implementation change or adding tests. We discussed PR stacks, that might not be right for us, but please make sure to call it out in PR description or add a comment on the PR for extra eyes on the public API changes.